bubblewrap.c

```diff
@@ -45,13 +45,29 @@
 static uid_t uid;
 static gid_t gid;
 static bool is_privileged;
+static bool did_setuid;
 static const char *argv0;
 static const char *host_tty_dev;
 static int proc_fd = -1;
 static char *opt_exec_label = NULL;
 static char *opt_file_label = NULL;
 static bool no_reaper;
 
+
+/* by default enabled caps for privileged user.  */
+#define BASE_CAPS0 (CAP_TO_MASK (CAP_NET_RAW) | CAP_TO_MASK (CAP_NET_BIND_SERVICE) | CAP_TO_MASK (CAP_AUDIT_WRITE) \
+                    | CAP_TO_MASK (CAP_DAC_OVERRIDE) | CAP_TO_MASK(CAP_SETFCAP) | CAP_TO_MASK(CAP_SETPCAP) \
+                    | CAP_TO_MASK (CAP_SETGID) | CAP_TO_MASK (CAP_SETUID) | CAP_TO_MASK (CAP_MKNOD) | CAP_TO_MASK (CAP_CHOWN) \
+                    | CAP_TO_MASK (CAP_FOWNER) | CAP_TO_MASK (CAP_FSETID) | CAP_TO_MASK (CAP_KILL) \
+                    | CAP_TO_MASK (CAP_SYS_CHROOT))
```

Comment:
Adding CAP_DAC_OVERRIDE here is vastly increasing the risk of having bubblewrap setuid. The whole reason we're doing this dance where we're only saving the caps we need for setting up the sandbox but running as the final user is that we can then be reasonably sure that the things that end up in the final sandbox is what the original user can access.

If the setup is run with DAC_OVERRIDE, then any source path you specifiy will be read with root privileges, which will let regular users access root-only files.